Tasks/DotNetCoreCLIV2/dotnetcore.ts

```diff
@@ -105,21 +108,43 @@ export class dotNetExe {
             } else {
                 dotnet.arg(projectFile);
             }
+            if(this.isBuildCommand()) {
+                var detailId = "";
+                var projectFilePath = path.dirname(projectFile);
+                if(this.logProjectEvents) {
+                    detailId = uuidV4();
+                    var projectFileName = path.basename(projectFile);
+                    var detailName = tl.loc("Build", projectFileName);
+                    var startTime = new Date(Date.now()).toISOString();
+                    //Todo: can we use named parameters in typescipt?
+                    tl.logDetail(detailId, "", "", "Process", detailName, 0, startTime, "", 0, tl.TaskState.Initialized);
```

Comment:
what would this do

Reply:
This creates a [timeline record](https://docs.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#logdetail-create-and-update-a-timeline-record-for-a-task) for the command, which is a recursive log according to the operations the command executes. This particular logger just prints the method names run as part of the command. It helps in debugging. This behaviour is present in msbuild and vsbuild tasks as well